### PR TITLE
[BTAT-5633] Controller gets mandation status, and changes front-end accordingly.

### PR DESCRIPTION
**Pull request by TheDeadlyPianist**

### CODE REVIEW

File: app/config/ConfigKeys.scala [hunk 1]
```diff
@@ -42,6 +42,7 @@ object ConfigKeys {
   val useDirectDebitDummyPageFeature: String = "features.useDirectDebitDummyPage.enabled"
   val useLanguageSelectorFeature: String = "features.useLanguageSelector.enabled"
   val submitReturnFeatures: String = "features.submitReturnFeatures.enabled"
+  val checkMandationStatus: String = "features.checkMandationStatus.enabled"
```
Comment (TomDudley92) [inline app/config/ConfigKeys.scala]: Please can you delete this as this feature switch already exists. It's the submitReturnFeatures one above.

File: app/controllers/VatDetailsController.scala [hunk 2]
```diff
@@ -54,33 +56,36 @@ class VatDetailsController @Inject()(val messagesApi: MessagesApi,
       for {
         customerInfo <- accountDetailsCall
         nextReturn <- returnObligationsCall
-        nextPayment <- if(retrieveHybridStatus(customerInfo)) Future.successful(Right(None)) else paymentObligationsCall
+        nextPayment <- if (retrieveHybridStatus(customerInfo)) Future.successful(Right(None)) else paymentObligationsCall
+        mandationStatus <- mandationStatusService.getMandationStatus(user.vrn)
```
Comment (TomDudley92) [inline app/controllers/VatDetailsController.scala]: Doesn't this need to be wrapped in the feature switch?

File: app/views/templates/nextReturnSection.scala.html [hunk 3]
```diff
@@ -38,7 +38,11 @@ <h2 class="heading-medium" style="margin-top: 0px">@messages("returnObligation.n
 
   <p>
     <a id="vat-returns-link" href="@{appConfig.vatReturnDeadlinesUrl}" data-journey-click="return:view-deadlines:vat-details">
-      @messages("returnObligation.viewReturns")
+        @if(isNonMTDfB) {
```
Comment (TomDudley92) [inline app/views/templates/nextReturnSection.scala.html]: Shouldn't this be wrapped in a feature switch?

File: conf/messages [hunk 4]
```diff
@@ -54,6 +54,7 @@ vatDetails.checkWhatYouOwe = Check what you owe and make a payment
 returnObligation.noReturn = No returns due right now
 returnObligation.nextReturn = Next return due
 returnObligation.viewReturns = View return deadlines
+returnObligation.submit = Submit return
```
Comment (TomDudley92) [inline conf/messages]: Do we need the Welsh Translation for this? Or will this be covered in another story?

File: app/views/templates/nextReturnSection.scala.html (hunk 3 above)
Comment (TheDeadlyPianist) [reply app/views/templates/nextReturnSection.scala.html]: The `isNonMTDfB` is the switch. Is this not how it's done?

Comment (TheDeadlyPianist) [reply app/views/templates/nextReturnSection.scala.html]: If the feature switch is set to false, then this value will always be false.

File: app/controllers/VatDetailsController.scala [hunk 5]
```diff
@@ -57,7 +57,8 @@ class VatDetailsController @Inject()(val messagesApi: MessagesApi,
         customerInfo <- accountDetailsCall
         nextReturn <- returnObligationsCall
         nextPayment <- if (retrieveHybridStatus(customerInfo)) Future.successful(Right(None)) else paymentObligationsCall
-        mandationStatus <- mandationStatusService.getMandationStatus(user.vrn)
+        mandationStatus <- if(appConfig.features.submitReturnFeatures()) {
+          mandationStatusService.getMandationStatus(user.vrn) } else { Future.successful(Right(MandationStatus("Disabled"))) }
```
Comment (TomDudley92) [inline app/controllers/VatDetailsController.scala]: This else hasn't been covered within the tests. Can a quick test be added to the Controller Spec just cover this please?

**Comment by TheDeadlyPianist** [comment]
@miranda-hawkes Assigned you to it because Tom told me to. It's the last comment he made that needs reviewing, and the Welsh content when I get it.

File: app/views/templates/nextReturnSection.scala.html (hunk 3 above)
Comment (miranda-hawkes) [inline app/views/templates/nextReturnSection.scala.html]: There's a scenario in the story that you've missed:
When we can't retrieve the mandation status at all (and feature switch is on), put the following text in the partial:
```
Sorry, there is a problem with the service. 
Try again later.
```
and don't show the 'Submit return' or 'Return deadlines' link. See the story and google doc for the full requirement

File: it/pages/VatDetailsPageSpec.scala [hunk 6]
```diff
@@ -41,15 +45,56 @@ class VatDetailsPageSpec extends IntegrationBaseSpec {
 
     "the user is authenticated" should {
 
-      "return 200" in new Test {
+      "return 200 and 'View returns deadline'" in new Test {
```
Comment (miranda-hawkes) [inline it/pages/VatDetailsPageSpec.scala]: Can you state 'and submit return feature switch is on' or something similar for these tests, and add (just one) scenario for when switch is off to prove the existing logic still stands. I know you tested that in the controller spec but if you're doing these scenarios at integration level you may as well add one with the switch off too

File: test/controllers/VatDetailsControllerSpec.scala [hunk 7]
```diff
@@ -180,12 +188,50 @@ class VatDetailsControllerSpec extends ControllerBaseSpec {
...
+            .returning(mandationStatusServiceResult).once()
         }
 
         val result: Future[Result] = target.details()(fakeRequest)
         status(result) shouldBe Status.OK
       }
     }
+
+    "the feature switch is turned off" should {
+      "return a VatDetailsViewModel as a MTDfB user" in new DetailsTest {
```
Comment (miranda-hawkes) [inline test/controllers/VatDetailsControllerSpec.scala]: Saying 'return a VatDetailsViewModel' etc doesn't really reflect what the tests are asserting

File: app/views/templates/nextReturnSection.scala.html (hunk 3 above)
Comment (TheDeadlyPianist) [reply app/views/templates/nextReturnSection.scala.html]: Oh, yeah. I opened the sub-task up, and didn't have the over-arching ticket open. My bad.

**Pull request merged**